docs/gds-supported-platform/tls-certificates.md

```diff
@@ -27,6 +32,76 @@ spec:
 
 > **Note:** cert-manager will need to be able to modify the DNS of the domains listed in the certificate in order to perform the DNS challenge. At the time of writing that only applies to the cluster domain.
 
+## Providing your own certificates
+
+If you want to provide your own certificate, you will need:
+
+ - a `SealedSecret` resource containing the certificate and key
+ - a `Gateway` resource defining the ingress domain
+
+We assume:
+
+ - you are in a namespace called `my-namespace`
+ - your certificate and key are in files called `my-cert.pem` and
+   `my-cert.key`
+ - you want to create a `SealedSecret` called `my-custom-cert` in a
+   file called `my-custom-cert.yaml`
+ - you are creating the certificate for the `sandbox` cluster
+ - the certificate is for `my-custom-domain.example.com`.
+
+To create a `SealedSecret` with your certificate, run:
+
+    kubectl create -n my-namespace secret generic my-custom-cert --dry-run --from-file=cert=my-cert.pem --from-file=key=my-cert.key --output yaml | gds sandbox seal --format yaml > my-custom-cert.yaml
+
+Here is an example of a `Gateway` that uses this certificate:
+
+```yaml
+apiVersion: networking.istio.io/v1alpha3
+kind: Gateway
+metadata:
+  name: custom-certificate-gateway
+  namespace: my-namespace
+  annotations:
+    externaldns.k8s.io/namespace: my-namespace
+spec:
+  selector:
+    istio: my-namespace-ingressgateway
+  servers:
+  - port:
+      number: 443
+      name: https
+      protocol: HTTPS
+    tls:
+      mode: SIMPLE
+      credentialName: my-custom-cert
+    hosts:
+      - "my-custom-domain.example.com"
+```
+
+The line `istio: my-namespace-ingressgateway` (replace `my-namespace`
+with your actual namespace name) selects the ingressgateway in your
+namespace.
+
+The line `credentialName: my-custom-cert` tells the `Gateway` to use
+the `SealedSecret` you created above for TLS termination.
+
+The line `- "my-custom-domain.example.com"` tells the `Gateway` to
+listen on the virtual host corresponding to the certificate domain
+name.
+
+### Custom domains
+
+As the example shows, if you provide your own certificates, you can
+use any custom domain you choose; you do not have to use the cluster
+`govsvc.uk` domain.  To use a custom domain, create a CNAME record
+from your domain to the external load balancer for your namespace.
```

Comment:
As a possible alternative to this...we may want to encourage an additional level of indirection: add a `.govsvc.uk` record and then have the custom domain CNAME to that. That way, if the `Service` that is responsible for the NLB gets recreated (we've seen that happen several times lately; granted more in Sandbox but it's happened twice in Verify) their "custom" CNAME will self-correct once `external-dns` does its job on the newly-created NLB.

Reply:
If we want to do this, we should either a) document how tenants can add a `.govsvc.uk` record, b) add a `.govsvc.uk` record ourselves.  I'm guessing you had one of these options in mind but I can't tell from your comment which one.  I think I'd prefer the second one.  How about `<namespace>-ingress.london.<cluster>.govsvc.uk`?  Or maybe plain `<namespace>.london.<cluster>.govsvc.uk`?

Reply:
Is https://github.com/alphagov/gsp/blob/master/docs/gds-supported-platform/external-dns.md not enough?

And I hadn't thought about adding one by default...If we do, of your suggestions, `<namespace>.<cluster-domain>` gets my vote.

Reply:
I've pushed an update that registers a standard DNS name for each ingressgateway